src/redprl/syntax.sml

```diff
@@ -15,6 +17,7 @@ struct
    | DFUN of 'a * variable * 'a | LAM of variable * 'a | AP of 'a * 'a
    | DPROD of 'a * variable * 'a | PAIR of 'a * 'a | FST of 'a | SND of 'a
    | ID_TY of (symbol * 'a) * 'a * 'a | ID_ABS of symbol * 'a | ID_AP of 'a * param
+   | HCOM of param list (* extents *) * dir * 'a (* type *) * 'a (* cap *) * (symbol * 'a) list (* tubes *)
```

Comment:
I don't think it's crucial for this PR; already, the syntax view is fairly fast-and-loose. We might make cleaning it up a separate task, rather than burdening you with it right now.